packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/utils/tests/hasContent.test.js

```diff
@@ -0,0 +1,66 @@
+import hasContent from '../hasContent';
+
+describe('hasContent', () => {
+  it('returns true for text content', () => {
+    const normalizedContent = hasContent('text', 'content');
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('returns false for empty text content', () => {
+    const normalizedContent = hasContent('text', '');
+    expect(normalizedContent).toEqual(false);
+  });
+
+  it('returns false for undefined text content', () => {
+    const normalizedContent = hasContent('text', undefined);
+    expect(normalizedContent).toEqual(false);
+  });
+
+  it('extracts content from single components with content', () => {
+    const normalizedContent = hasContent(
+      'component',
+      { name: 'content' },
+      { mainField: { name: 'name' } }
+    );
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('extracts content from single components without content', () => {
+    const normalizedContent = hasContent(
+      'component',
+      { name: '' },
+      { mainField: { name: 'name' } }
+    );
+    expect(normalizedContent).toEqual(false);
+  });
+
+  it('extracts content from repeatable components with content', () => {
+    const normalizedContent = hasContent(
+      'component',
+      [{ name: 'content_2', value: 'truthy' }],
+      { mainField: { name: 'content_2' } },
+      { repeatable: true }
+    );
+    expect(normalizedContent).toEqual(true);
+  });
+
+  it('extracts content from repeatable components without content', () => {
+    const normalizedContent = hasContent(
+      'component',
+      [{ name: 'content_2', value: '' }],
```

Comment:
Components always have an `id` field.

Reply:
Fixed in 44921d68c2f3b94d626f91dc769aa255481dbd74